src/apps/Permissions/AppRoles.js

```diff
@@ -98,10 +98,12 @@ class RoleRow extends React.Component {
         <AppInstanceLabel app={manager.app} proxyAddress={manager.address} />
       )
     }
-    if (manager.type === 'burn') {
-      return <IdentityBadge entity={'Discarded'} />
-    }
-    return <IdentityBadge entity={manager.address} />
+    return (
+      <LocalIdentityBadge
+        address={manager.address}
+        entity={manager.type === 'burn' ? 'Discarded' : manager.address}
```

Comment:
In particular, resolving to an address here if the `manage.type === 'burn'` is the wrong behaviour. This shouldn't need to resolve to another identity.

Reply:
Not sure I understand. What does `manage.type === 'burn'` mean? and should we just render a normal `IdentityBadge` when we want to show `Discarded`?

Reply:
> Should we just render a normal IdentityBadge when we want to show Discarded?

Yes; these "Any" and "Discarded" ("burned") entities are special sentinel values we use in the smart contracts to denote a specific case in their logic. They are not real addresses, and should never be resolved.